### PR TITLE
`fix`: Made a change for the ldbc test which will fail the test if file download fails.

### PR DESCRIPTION
**Pull request by SiddeshUbale**
`fix`: Made a change for the ldbc test which will fail the test if file download fails.

### CODE REVIEW

File: t/t.go [hunk 1]
```diff
@@ -1006,9 +1011,6 @@ func run() error {
 		if testSuiteContains("load") || testSuiteContains("all") {
 			downloadDataFiles()
```
Comment (mangalaman93) [inline t/t.go:1012]: will we have the same problem if this function fails too?

File: t/t.go [hunk 2]
```diff
@@ -1050,10 +1052,31 @@ func validateAllowed(testSuite []string) {
 	}
 }
 
+func validateDownload() {
+	// 	if len(*download) == 0 {
```
Comment (mangalaman93) [inline t/t.go:1056]: still need these?

File: t/t.go [hunk 3]
```diff
@@ -1050,10 +1052,31 @@ func validateAllowed(testSuite []string) {
...
+	// 	}
+	// 	if *download != "ldbc" {
+	// 		log.Fatalf("Only ldbc is supported for download")
+	// 	}
+	if testSuiteContains("ldbc") || testSuiteContains("all") {
+		out, err := downloadLDBCFiles()
+		length := len(out)
+		for i := 0; i < length; i++ {
+			str := string(out[i])
+			log.Println("Error Downloading a file: ", str)
```
Comment (mangalaman93) [inline t/t.go:1067]: just use small letters in panic comment, same below

File: t/t.go (hunk 1 above)
Comment (SiddeshUbale) [reply t/t.go:1020]: Resolved: Applied the same fix for the load test.

File: t/t.go (hunk 2 above)
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go (hunk 3 above)
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go [hunk 4]
```diff
@@ -802,11 +807,16 @@ func downloadDataFiles() {
 		if out, err := cmd.CombinedOutput(); err != nil {
 			fmt.Printf("Error %v", err)
 			fmt.Printf("Output %v", out)
+			outInfo = append(outInfo, out)
+			errInfo = append(errInfo, err)
 		}
 	}
+	return outInfo, errInfo
```
Comment (mangalaman93) [inline t/t.go:814]: can we check for error right here instead of returning it?

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: t/t.go [hunk 5]
```diff
@@ -786,7 +786,12 @@ var ldbcDataFiles = map[string]string{
 	"ldbcTypes.schema": "https://github.com/dgraph-io/benchmarks/blob/master/ldbc/sf0.3/ldbcTypes.schema?raw=true",
 }
 
-func downloadDataFiles() {
+var outInfo [][]byte
```
[the remark this answers is not in the document]
Comment (SiddeshUbale) [reply t/t.go:789]: All the variables related to ldbc and load datafiles are declared here, that's why I appended both of these variables right below them, We can declare them at the top of the file.

File: t/t.go (hunk 4 above)
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go [hunk 6]
```diff
@@ -800,8 +800,10 @@ func downloadDataFiles() {
 		cmd.Dir = *tmp
 
 		if out, err := cmd.CombinedOutput(); err != nil {
-			fmt.Printf("Error %v", err)
-			fmt.Printf("Output %v", out)
+			fmt.Printf("Error %v /n", err)
+			str := string(out)
```
Comment (mangalaman93) [inline t/t.go:804]: we don't need to do this assignment here, we can directly print it

File: t/t.go [hunk 7]
```diff
@@ -800,8 +800,10 @@ func downloadDataFiles() {
...
+			log.Println("error downloading a file: ", str)
+			log.Panic("error downloading required load test files")
```
Comment (mangalaman93) [inline t/t.go:806]: why don't we just do `panic("error downloading required load test files")` here instead?

File: t/t.go [hunk 8]
```diff
@@ -800,8 +800,10 @@ func downloadDataFiles() {
 		cmd.Dir = *tmp
 
 		if out, err := cmd.CombinedOutput(); err != nil {
-			fmt.Printf("Error %v", err)
-			fmt.Printf("Output %v", out)
+			fmt.Printf("Error %v /n", err)
+			str := string(out)
+			log.Println("error downloading a file: ", str)
```
Comment (mangalaman93) [inline t/t.go:805]: we should just use fmt everywhere.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]
A few more comments, this looks better. Could you show an example output too?

File: t/t.go (hunk 6 above)
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go (hunk 8 above)
Comment (SiddeshUbale) [reply t/t.go]: Resolved.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: t/t.go [hunk 9]
```diff
@@ -791,17 +791,13 @@ func downloadDataFiles() {
...
-	}
-	x.Check(testutil.MakeDirEmpty([]string{*tmp}))
 	for fname, link := range datafiles {
 		cmd := exec.Command("wget", "-O", fname, link)
 		cmd.Dir = *tmp
 
 		if out, err := cmd.CombinedOutput(); err != nil {
-			fmt.Printf("Error %v", err)
-			fmt.Printf("Output %v", out)
+			fmt.Printf("Error %v /n", err)
```
[the remark this answers is not in the document]
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go [hunk 10]
```diff
@@ -833,7 +824,7 @@ func downloadLDBCFiles() {
 			cmd.Dir = *tmp
 			if out, err := cmd.CombinedOutput(); err != nil {
 				fmt.Printf("Error %v", err)
```
[the remark this answers is not in the document]
Comment (SiddeshUbale) [reply t/t.go]: Resolved

File: t/t.go [hunk 11]
```diff
@@ -1003,11 +994,21 @@ func run() error {
 	go func() {
 		defer close(testCh)
 		valid := getPackages()
-		if testSuiteContains("load") || testSuiteContains("all") {
-			downloadDataFiles()
-		}
 		if testSuiteContains("ldbc") || testSuiteContains("all") {
+			if *tmp == "" {
```
[the remark this answers is not in the document]
Comment (SiddeshUbale) [reply t/t.go]: Resolved

[the remark this answers is not in the document]
Comment (mangalaman93) [reply t/t.go:999]: We can move the following code outside the ifs and before them:
```
if *tmp == "" {
    *tmp = filepath.Join(os.TempDir(), "/ldbcdata")
}
x.Check(testutil.MakeDirEmpty([]string{*tmp}))
```

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

**Review by mangalaman93 — APPROVED** [review]

**Review by all-seeing-code — APPROVED** [review]

**Pull request merged**